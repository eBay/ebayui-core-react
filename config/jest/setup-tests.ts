import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
