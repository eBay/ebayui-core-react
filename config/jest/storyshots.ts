import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
