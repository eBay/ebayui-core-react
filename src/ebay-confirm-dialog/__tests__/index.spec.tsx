import React from 'react'
import requireContext from 'node-require-context'
import { render, fireEvent, RenderResult } from '@testing-library/react'
import { EbayDialogHeader } from '../..'
import { EbayConfirmDialog } from '../index';
import { initStoryshots } from '../../../config/jest/storyshots';

jest.mock('../../common/random-id', () => ({ randomId: () => 'abc123' }))


describe('<EbayConfirmDialog>', () => {
    let wrapper: RenderResult
    const closeDrawerHandler = jest.fn()

    beforeEach(() => {
        wrapper = render(
            <EbayConfirmDialog
                animated={false}
                confirmText="Confirm"
                rejectText="Reject"
                open
                onReject={closeDrawerHandler}
            >
                <EbayDialogHeader>Heading</EbayDialogHeader>
                <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                    incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
                    nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </p>
                <p><a href="http://www.ebay.com">www.ebay.com</a></p>
                <input placeholder="Enter a value" />
            </EbayConfirmDialog>
        )
    })

    it('should have close button', () => {
        expect(wrapper.getByText('Reject')).toBeInTheDocument()
    })

    it('should focus on Okay button', () => {
        expect(wrapper.getByText('Confirm')).toHaveFocus()
    })

    it('should trigger onClose when close button is clicked', () => {
        fireEvent.click(wrapper.getByText('Reject'))
        expect(closeDrawerHandler).toBeCalled()
    })
})

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/);
        return configure(req, module)
    }
});
