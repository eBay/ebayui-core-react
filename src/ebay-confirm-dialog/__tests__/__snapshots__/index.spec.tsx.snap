// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-confirm-dialog Default 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
  >
    <button
      aria-hidden="true"
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p
      aria-hidden="true"
    >
      Some outside content...
    </p>
    <div
      aria-hidden="false"
    >
      <div
        aria-hidden="false"
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="confirm-dialog confirm-dialog--mask-fade confirm-dialog--show-init"
        role="dialog"
      >
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
        <div
          aria-hidden="false"
          class="confirm-dialog__window confirm-dialog__window--fade keyboard-trap--active"
        >
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
          <div
            class="confirm-dialog__header"
          >
            <h2
              id="dialog-title-abc123"
            >
              Delete Address?
            </h2>
          </div>
          <div
            class="confirm-dialog__main"
            id="confirm-dialog-main"
          >
            <p>
              You will permanently lose this address.
            </p>
          </div>
          <div
            class="confirm-dialog__footer"
          >
            <button
              class="btn confirm-dialog__reject btn--secondary"
            >
              Delete
            </button>
            <button
              aria-describedby="confirm-dialog-main"
              class="btn confirm-dialog__confirm btn--primary"
              id="confirm-dialog-confirm"
            >
              Cancel
            </button>
          </div>
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
        </div>
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-confirm-dialog With Animation 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p>
      Some outside content...
    </p>
    <div>
      <div
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="confirm-dialog confirm-dialog--mask-fade"
        hidden=""
        role="dialog"
      >
        <div
          class="confirm-dialog__window confirm-dialog__window confirm-dialog__window--fade"
        >
          <div
            class="confirm-dialog__header"
          >
            <h2
              id="dialog-title-abc123"
            >
              Delete Address?
            </h2>
          </div>
          <div
            class="confirm-dialog__main"
            id="confirm-dialog-main"
          >
            <p>
              You will permanently lose this address.
            </p>
          </div>
          <div
            class="confirm-dialog__footer"
          >
            <button
              class="btn confirm-dialog__reject btn--secondary"
            >
              Delete
            </button>
            <button
              aria-describedby="confirm-dialog-main"
              class="btn confirm-dialog__confirm btn--primary"
              id="confirm-dialog-confirm"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
